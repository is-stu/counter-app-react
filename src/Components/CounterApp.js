import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ value = 10 }) => {
  const [counter, setCounter] = useState(value);

  const handleSubstract = () => {
    setCounter(counter - 1);
  };

  const handleReset = () => {
    setCounter((value));
  };
  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>
      <button
        onClick={() => {
          setCounter(counter + 1);
        }}>
        Sumar 1
      </button>
      <button onClick={handleReset}>Reset</button>
      <button onClick={handleSubstract}>Restar 1</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number,
};
