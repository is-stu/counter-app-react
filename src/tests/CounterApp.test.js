import { shallow } from 'enzyme';
import React from 'react';
import { CounterApp } from '../Components/CounterApp';

describe('Pruebas en el counterApp', () => {
  let wrapper = shallow(<CounterApp />);

  beforeEach(() => {
    wrapper = shallow(<CounterApp />);
  });

  test('debe mostrar counterApp correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de mostrar el valor por defecto de 100', () => {
    const wrapper = shallow(<CounterApp value={100} />);
    const counterText = wrapper.find('h2').text().trim();
    expect(counterText).toBe('100');
  });

  test('debe incrementar contador', () => {
    wrapper.find('button').at(0).simulate('click');
    const counterText = wrapper.find('h2').text().trim();
    expect(counterText).toBe('11');
  });

  test('debe decrementar con el boton -1', () => {
    wrapper.find('button').at(2).simulate('click');
    const counterText = wrapper.find('h2').text().trim();
    expect(counterText).toBe('9');
  });

  test('prueba boton reiniciar', () => {
    const wrapper = shallow(<CounterApp value={105} />);
    wrapper.find('button').at(0).simulate('click');
    wrapper.find('button').at(1).simulate('click');
    const counterText = wrapper.find('h2').text().trim();
    expect(counterText).toBe('105');
  });
});
