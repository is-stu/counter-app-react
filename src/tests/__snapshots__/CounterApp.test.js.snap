// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el counterApp debe mostrar counterApp correctamente 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    Sumar 1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    Restar 1
  </button>
</Fragment>
`;
